src/index.js

```diff
@@ -62,40 +62,37 @@ You can find instructions for installing Rust and Cargo at:
 
 function processCargoBuildLine(options, copied, line) {
   const data = JSON.parse(line);
+  const { filenames, reason, target } = data;
 
-  if (!data || data.reason !== "compiler-artifact" || !data.target) {
+  if (!data || reason !== "compiler-artifact" || !target) {
     return;
   }
 
   const { kind: kinds, name } = data.target;
 
-  if (!Array.isArray(kinds) || !kinds.length) {
+  if (!Array.isArray(kinds) || !Array.isArray(filenames)) {
     return;
   }
 
-  const [kind] = kinds;
-  const key = getArtifactName({ artifactType: kind, crateName: name });
-  const outputFiles = options.artifacts[key];
+  kinds.forEach((kind, i) => {
```

Comment:
It'd probably help to write a short comment here that the `kind` and `filenames` fields of the JSON match up pairwise.